src/configuration/common/raw_configuration.hpp

```diff
@@ -27,13 +27,26 @@ class raw_configuration : public ddwaf_object {
 
     raw_configuration() = default;
     // NOLINTNEXTLINE(google-explicit-constructor,hicpp-explicit-conversions)
-    raw_configuration(const ddwaf_object &arg) : _ddwaf_object() { *((ddwaf_object *)this) = arg; }
+    raw_configuration(const ddwaf_object &arg) : _ddwaf_object()
+    {
+        dynamic_cast<ddwaf_object &>(*this) = arg;
```

Comment:
dynamic_cast?? First, this could be a `static_cast` since it's an upcast; second, can't this be written just as

```
raw_configuration(const ddwaf_object &arg) : _ddwaf_object{arg} {}
```

Reply:
Not sure what I was thinking here